### PR TITLE
refactor(migrations): Add optional template reformatting after control flow migration

### PR DESCRIPTION
**Pull request by jessicajaniuk**
This adds the ability to reformat the template with proper indents after the migraion finishes. It is an optional parameter.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/schematics/ng-generate/control-flow-migration/util.ts [hunk 1]
```diff
@@ -409,6 +409,64 @@ export function getMainBlock(etm: ElementToMigrate, tmpl: string, offset: number
...
+ */
+export function formatTemplate(tmpl: string): string {
+  if (tmpl.indexOf('\n') > -1) {
+    // match any type of control flow block as start of string ignoring whitespace
+    // @if | @switch | @case | @default | @for | } @else
+    const openBlockRegex = /^\s*\@(if|switch|case|default|for)|^\s*\}\s\@else/;
+
+    // regex for matching an html element opening
+    // <div thing="stuff" [binding]="true"> || <div thing="stuff" [binding]="true"
+    const openElRegex = /^\s*<([a-z]+)(?![^>]*\/>)[^>]*>?/;
```
Comment (crisbeto) [inline packages/core/schematics/ng-generate/control-flow-migration/util.ts:423]: Should this account for self-closing tags?

**Review by crisbeto — APPROVED** [review]

File: packages/core/schematics/ng-generate/control-flow-migration/util.ts (hunk 1 above)
Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/util.ts:423]: It should already account for that situation, but I'll add a test. One case that it doesn't cover is input tags on multiple lines that don't have the self closing `/`. That's a tough one to handle without more complicated logic.

File: packages/core/schematics/ng-generate/control-flow-migration/schema.json [hunk 2]
```diff
@@ -9,6 +9,12 @@
       "description": "Path relative to the project root which should be migrated",
       "x-prompt": "Which path in your project should be migrated?",
       "default": "./"
+    },
+    "format": {
+      "type": "boolean",
+      "description": "Enables reformatting of your templates",
+      "x-prompt": "Should the migration reformat your templates?",
+      "default": "false"
```
[the remark this answers is not in the document]
Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/schema.json:17]: I go back and forth on that. I'll submit this as is, but we can discuss during standup.

File: packages/core/schematics/ng-generate/control-flow-migration/util.ts [hunk 3]
```diff
@@ -409,6 +409,64 @@ export function getMainBlock(etm: ElementToMigrate, tmpl: string, offset: number
...
 }
 
+/**
+ * re-indents all the lines in the template properly post migration
+ */
+export function formatTemplate(tmpl: string): string {
+  if (tmpl.indexOf('\n') > -1) {
+    // match any type of control flow block as start of string ignoring whitespace
+    // @if | @switch | @case | @default | @for | } @else
+    const openBlockRegex = /^\s*\@(if|switch|case|default|for)|^\s*\}\s\@else/;
```
[the remark this answers is not in the document]
Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/util.ts:419]: This currently looks for `} @else` and ignores whatever comes after it. That should include `@else if`, but I'll add a test to verify.

File: packages/core/schematics/ng-generate/control-flow-migration/util.ts (hunk 1 above)
Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/util.ts:423]: Test added and passed. 🟢

File: packages/core/schematics/ng-generate/control-flow-migration/util.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jessicajaniuk) [reply packages/core/schematics/ng-generate/control-flow-migration/util.ts:419]: Test added and 🟢

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 6e29e853ed6973774acfe55c41da691d430bf54b.

**Pull request closed without merging**